README.md

```diff
@@ -31,6 +31,11 @@ An `$ExpectType` directive tests the type of the expression on the next line. Th
     npm install -g typings-checker
     typings-checker your-test.ts
 
+## options
+
+* `--noLines` (`-l`): hide line numbers in output -- useful if you don't want these clogging diff logs when you add or remove a line.
```

Comment:
Filtering out the line numbers that way seems like an interesting alternative to me. It hadn't occurred to me, though I suppose I'd felt the option might help people in a similar situation -- which I hope wouldn't be many. Including the code fragments was just a way to make it useful even without line numbers.

Zero assertion failures sounds wonderful to me. I do want that, but that's not where we are for ramda right now; there are quite some blocking issues preventing me from reaching that point even if I wanted to: [erased generics](https://github.com/Microsoft/TypeScript/issues/9949), lack of support for [heterogeneous `map()`](https://github.com/Microsoft/TypeScript/issues/12393), lack of type-level support for [`reduce()`](https://github.com/Microsoft/TypeScript/issues/12512), quite possibly support for [higher-kinded types](https://github.com/Microsoft/TypeScript/issues/1213), not to mention whatever [other](https://github.com/Microsoft/TypeScript/issues/6606) features this list might depend on, aside from just finding the time to figure out every single step between here and proper type inference.

I mean, TypeScript has 1,800+ outstanding issues at the moment, so in that sense, I fear it's no surprise to find that not everything is quite perfect yet. FP libraries are probably among the harder ones to type, but I'm probably also being more ambitious than e.g. the lodash typings -- if I map over an object, I'd like for my IDE to know both the (distinct) types and keys of each value. Settling for vague types is an easy way to pass all tests (as you noted), but in a sense feels to me like it beats the point of doing a typings library.

To me this was also the reason to commit and diff the output, like you were. Up to now I had no idea if an edit would break inference and turned everything into `any`. With your library, now I do. But I need to be able to answer bigger questions than that; notably: given my current assertion failures (whether `any` or errors), would switching to an [experimental branch](https://github.com/Microsoft/TypeScript/issues/9949) make that overall situation better or worse?

I also hope I can get to the point of having the luxury to get things fixed. And I'm about to find out more about that -- knowing what's good or bad now, I can finally start seeing more about that.